### PR TITLE
feat(dop): R&D workflow settings optimization

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
R&D workflow settings optimization

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMjk3XSwiYXNzaWduZWUiOlsiMTAwMTI2MSJdfQ%3D%3D&id=321413&iterationID=1297&pId=319942&type=TASK)

#### Specified Reviewers:

/assign @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     R&D workflow settings optimization         |
| 🇨🇳 中文    |       研发工作流设置优化       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: api/proto/dop/devflowrule/devflowrule.proto [hunk 1]
```diff
@@ -82,31 +83,45 @@ message GetDevFlowRuleResponse {
...
     repeated Flow flows = 2 ;
-    uint64 orgID = 3;
-    string orgName = 4;
-    uint64 projectID = 5;
-    string projectName = 6;
-    google.protobuf.Timestamp timeCreated = 7;
-    google.protobuf.Timestamp timeUpdated = 8;
-    string creator = 9;
-    string updater = 10;
+    repeated BranchPolicy branchPolicies = 3;
```
Comment (sfwn) [inline api/proto/dop/devflowrule/devflowrule.proto:86]: for proto, you should add new field to the end

File: apistructs/devflowrule.go [hunk 2]
```diff
@@ -24,68 +24,75 @@ type DevFlowRuleResponse struct {
...
-	Place            string
-	ChangeBranchRule string
-}
+type (
+	BranchPolicy struct {
+		Branch     string   `json:"branch"`
+		BranchType string   `json:"branchType"`
+		Policy     *BPolicy `json:"policy"`
+	}
+	BPolicy struct {
```
Comment (sfwn) [inline apistructs/devflowrule.go:53]: what is BPolicy means

File: internal/apps/dop/providers/devflowrule/devflowrule.service.go [hunk 3]
```diff
@@ -68,76 +88,72 @@ func (p *provider) CreateDevFlowRule(ctx context.Context, request *pb.CreateDevF
...
-				{
-					Place:            "TASK",
-					ChangeBranchRule: "feature/*",
-				},
-				{
-					Place:            "BUG",
-					ChangeBranchRule: "bugfix/*",
-				},
-			},
+			Name:         "DEV",
```
Comment (sfwn) [inline internal/apps/dop/providers/devflowrule/devflowrule.service.go:108]: should use const or vars of alias-type

**Review by sfwn — CHANGES_REQUESTED** [review]

File: api/proto/dop/devflowrule/devflowrule.proto (hunk 1 above)
Comment (littlejiancc) [reply api/proto/dop/devflowrule/devflowrule.proto:86]: done

File: apistructs/devflowrule.go (hunk 2 above)
Comment (littlejiancc) [reply apistructs/devflowrule.go:53]: done

**Review by sfwn — APPROVED** [review]

**Pull request merged**